io.github.RobertBeckebans.RBDoom3BFG.yaml

```diff
@@ -0,0 +1,133 @@
+    build-options:
+      env:  
+        CXXFLAGS: -fvisibility-inlines-hidden -fvisibility=hidden
+    sources:
+      - type: git
+        url: https://github.com/microsoft/DirectXShaderCompiler
+        tag: v1.8.2405
+    post-install:
+      # Install binaries
+      - install -Dm 755 bin/dxc /app/bin/dxc
+      - install -Dm 755 lib/libdxcompiler.so /app/lib/libdxcompiler.so
+      # Install headers
+      - install -D ../include/dxc/dxcapi.h /app/include/dxc/dxcapi.h
+      - install -D ../include/dxc/dxcerrors.h /app/include/dxc/dxcerrors.h
+      - install -D ../include/dxc/dxcisense.h /app/include/dxc/dxcisense.h
+      - install -D ../include/dxc/WinAdapter.h /app/include/dxc/WinAdapter.h
+
+  # build application from Git sources
+  - name: rbdoom3bfg-build
+    buildsystem: cmake-ninja
+    builddir: true
+    subdir: neo
+    no-make-install: true
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DFFMPEG=Off
+      - -DBINKDEC=On
+      - -DRETAIL=Off
+      - -DFLATPAK=On
+      - -DUSE_PRECOMPILED_HEADERS=Off
+      - -DUSE_INTRINSICS_SSE=Off
+      - -DUSE_INTRINSICS_NEON=Off
+    build-options:
+      arch:
+        x86_64:
+          config-opts:
+            - -DUSE_INTRINSICS_SSE=On
+        aarch64:
+          config-opts:
+            - -DUSE_INTRINSICS_SSE=Off
+    sources:
+      - type: git
+        url: https://github.com/RobertBeckebans/RBDOOM-3-BFG
+        #branch: flatpak
+        commit: 5d6216b93a3cd194797271e17533bda81b57f142
+      - type: patch
+        path: shader-compiler.patch
+      - type: patch
+        path: shader-make.patch
+    post-install:
+      # install executable
+      - install -Dm 755 RBDoom3BFG /app/bin/RBDoom3BFG
+      # install base files
+      - mkdir -p /app/share/rbdoom3bfg/base
+      - cp ../../base/default.cfg /app/share/rbdoom3bfg/base/default.cfg
+      - cp -R ../../base/renderprogs2 /app/share/rbdoom3bfg/base/
+      # install desktop files
+      - install -Dm 644 ../../install/desktop/RBDoom3BFG.metainfo.xml /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      - install -Dm 644 ../../install/desktop/RBDoom3BFG.desktop /app/share/applications/${FLATPAK_ID}.desktop
+      - install -Dm 644 ../../install/desktop/RBDoom3BFG.png /app/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png
+
+  # module to install p7zip
+  - name: p7zip
+    sources:
+      - type: archive
+        url: https://github.com/p7zip-project/p7zip/archive/refs/tags/v17.05.tar.gz
+        sha256: d2788f892571058c08d27095c22154579dfefb807ebe357d145ab2ddddefb1a6
+    buildsystem: cmake-ninja
+    subdir: CPP/7zip/CMAKE/7zr
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+    no-make-install: true
+    post-install:
+      - install -Dm 755 7zr /app/bin/7zr
+
+  # install .pk4 files from Windows version
+  - name: rbdoom3bfg-assets
+    buildsystem: simple
+    #cleanup: ['*']  # Remove extracted files after installation
+    sources:
+      - type: file
```

Comment:
use it as archive and drop p7zip?

Reply:
I tried the archive method but it always messed up the temporal unpacking and quit with an error of a duplicated name file or so because that file is in the original Windows 7z release multiple times. Like mod_ammo/run_mod.bat and mod_ultimateHD/run_mod.bat were conflicting files. Building p7zip was the only way to unpack the correct files and I hesitate to upload the release to github another time using tar.gz.